gammapy/background/models.py

```diff
@@ -650,3 +651,26 @@ def compute_rate(self):
         bg_rate = bg_rate.to('MeV-1 sr-1 s-1')
 
         self.bg_rate.data = bg_rate
+
+    def smooth(self):
+        """Smooth the bkg rate with a gaussian 1D kernel.
+
+        """
+        for binE, E in enumerate(self.counts.energy[:-1]):
+            counts = self.counts.data[binE, :]
+            Nev = np.sum(counts).value
+            if (Nev > 0):
+                Np = len(counts)
+                # Number of pixels per sigma of the kernel gaussian to have more than 150 events/sigma
+                Npix_sigma = (150 / Nev) * Np
+                # For high statistic, we impose a minimum of 4pixel/sigma
+                Npix_sigma = np.maximum(Npix_sigma, 4)
+                # For very low statistic, we impose a maximum lenght of the kernel equal of the number of bin
+                # in the counts histogram
+                Npix_sigma = np.minimum(Npix_sigma, Np / 6)
+                # kernel gaussian define between -3 and 3 sigma
+                x = np.linspace(-3, 3, 6 * Npix_sigma)
```

Comment:
When you care about speed, by far the best option would be to use `scipy.ndimage.filters.gaussian_filter`.